services/ymax-planner/src/engine.ts

```diff
@@ -674,9 +653,11 @@ export const startEngine = async (
     });
 
     // Process existing pending transactions on startup
-    void handlePendingTx({ ...evmCtx, signingSmartWalletKit, fetch }, tx, {
-      log,
-    }).catch(logIgnoredError);
+    void handlePendingTx(
+      { ...evmCtx, signingSmartWalletKit, fetch, cosmosRest },
```

Comment:
I'd like to express my frustration with `EvmContext`. Consistency-ensuring field overrides like this jump out as indicative of an incoherent design.